Fix wording around token v2 royalty (#8438)
aptos-labs/aptos-core: developer-docs-site/docs/standards/aptos-token-v2.md

* Soul bound tokens can be easily supported

## Collections and tokens as objects
In this Token standard, both collections and tokens will be separate [objects](./aptos-object.md). They have their own
distinct addresses and can be referenced both on and off chain by address. Each object can contain multiple resources
so collections and tokens are extensible by default, allowing the creator to add custom data and functionalities without
having to modify the framework.

On chain, another struct can include a reference to the collection or token objects like below:
```rust
struct ReferenceExample has key {
    my_collection: Object<Collection>,
    my_token: Object<Token>,
}
```
where both `my_collection` and `my_token` are addresses (with `Object<>` wrapper).

Off-chain, the address of an object can be passed along to create a transaction that calls entry functions with `Object<T>`
as arguments. For example:
```rust
public entry fun my_function(my_collection: Object<Collection>) {
    // Do something with the collection
}
```

Collection and token addresses will also be used to query data such as fetching all resources via fullnode API or against
an indexing service.

### Royalties

> Royalties are simply [a resource with associated functionality](https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/aptos-token-objects/sources/royalty.move)

Reviewer: When you mention "associated functionality", do you mean a "module"?

Author: Yep updated to make this clearer

* NFTs can own other NFTs adding easy composability
* Soul bound tokens can be easily supported

## Collections and tokens as objects
In this Token standard, both collections and tokens will be separate [objects](./aptos-object.md). They have their own
distinct addresses and can be referenced both on and off chain by address. Each object can contain multiple resources
so collections and tokens are extensible by default, allowing the creator to add custom data and functionalities without
having to modify the framework.

On chain, another struct can include a reference to the collection or token objects like below:
```rust
struct ReferenceExample has key {
    my_collection: Object<Collection>,
    my_token: Object<Token>,
}
```
where both `my_collection` and `my_token` are addresses (with `Object<>` wrapper).

> Off-chain, the address of an object can be passed along to create a transaction that calls entry functions with `Object<T>`

Reviewer 2: ```suggestion
Off-chain, the address of the object can be passed along to replace object arguments in entry functions called by transaction creation.
```

* Royalty - specifies how many % of the sale price goes to the creator of the collection. This can be changed with a
`MutatorRef` generated by the Royalty module.

A `MutatorRef`, a storable struct that grants permissions to mutate, can be generated only during creation of the collection.

> If created, the holder of the `MutatorRef` can change the `description` and the `URI length` of the collection.

Author: Yes that's generic Move material

}

public entry fun create_collection(creator: &signer) {
    // Constructor ref is a non-storable struct returned when creating a new object.

>     // It can be exchanged for signer to add resources to the collection object.

Reviewer 2: ```suggestion
    // It can generate an object signer to add resources to the collection object.
```

}
```

Creators should cautiously consider whether they should use `create_named_token` or `create_from_account` when building
their custom collection/token. In general `create_from_account` is recommended as it allows for clean deletion if the

> tokens are burnt and generally, deterministic addresses for tokens are not always necessary thanks to indexing services.

Reviewer: I don't completely understand what "clean deletion" means. And why `create_from_account` supports clean deletion but not `create_named_token`?

Reviewer: If `create_from_account` is strictly better, then what's the advantage of using `create_named_token`?

If a creator wants more custom functionalities such as being able to forcefully transfer a soul bound token, they would
need to write their own custom module that builds on top of the base token v2 standard. They can of course borrow inspiration
and code from the Aptos Token module.

## Fungible Token
Similar to [EIP-1155](https://eips.ethereum.org/EIPS/eip-1155), the Token v2 standard also supports fungible tokens

> (also known as semi-fungible tokens). An example of this would be armor tokens in a game. Each armor token represents a

Reviewer 2: fungible, semi-fungible? ...so confusing.

Author: These are common terms though
